src/main/java/frc/robot/commands/ShooterPressed.java

```diff
@@ -40,35 +42,36 @@ public void initialize() {
   @Override
   public void execute() {
     m_ShooterSubsystem.prime();
-    if (m_ShooterSubsystem.wheelReady()) {
-      i++;
-      // 1000 miliseconds delay
-      if (i == 50) {
-        m_ShooterSubsystem.runCargo(true, true);
-        m_ShooterSubsystem.SetCargoBoolean(true);
-      }
-
-    } else {
-      m_ShooterSubsystem.runCargo(false, false);
-    }
-    m_ShooterSubsystem.UpdateIdelay(i);
   }
 
   // Called once the command ends or is interrupted.
   @Override
   public void end(boolean interrupted) {
     m_ShooterSubsystem.runCargo(false, false);
     m_ShooterSubsystem.windFlywheel(0);
-    m_ShooterSubsystem.SetCargoBoolean(false);
-    m_ShooterSubsystem.UpdateIdelay(0);
+    m_ShooterSubsystem.setCargoBoolean(false);
+    m_ShooterSubsystem.updateIdelay(0);
 
     // SmartDashboard.putBoolean("wheelReady", false);
   }
 
   // Returns true when the command should end.
   @Override
   public boolean isFinished() {
+    if (m_ShooterSubsystem.wheelReady()) {
+      if (i > 0 || !LLEnabled || m_LimelightSubsystem.calculatePID() == 0.0) {
+        i++;
+        m_ShooterSubsystem.runCargo(true, true);
+        m_ShooterSubsystem.setCargoBoolean(true);
+        if (i >= 50) {// 1000 miliseconds delay
```

Comment:
Same comment as from ShooterHeld -- if this is to make sure we've shot all the cargo, polling the CDS sensors could be more accurate